uxarray/core/grid.py

```diff
@@ -123,30 +123,30 @@ def __init_grid_var_attrs__(self) -> None:
         """
 
         # Iterate over dict to set access attributes
-        for key, value in self.ds_var_names.items():
+        for key, value in self.grid_var_names.items():
             # Set Attributes for Data Variables
-            if self.ds.data_vars is not None:
-                if value in self.ds.data_vars:
-                    setattr(self, key, self.ds[value])
+            if self._ds.data_vars is not None:
+                if value in self._ds.data_vars:
+                    setattr(self, key, self._ds[value])
 
             # Set Attributes for Coordinates
-            if self.ds.coords is not None:
-                if value in self.ds.coords:
-                    setattr(self, key, self.ds[value])
+            if self._ds.coords is not None:
+                if value in self._ds.coords:
+                    setattr(self, key, self._ds[value])
 
             # Set Attributes for Dimensions
-            if self.ds.dims is not None:
-                if value in self.ds.dims:
-                    setattr(self, key, len(self.ds[value]))
+            if self._ds.dims is not None:
+                if value in self._ds.dims:
+                    setattr(self, key, len(self._ds[value]))
 
     def __from_vert__(self):
         """Create a grid with one face with vertices specified by the given
         argument.
 
         Called by :func:`__init__`.
         """
-        self.ds = xr.Dataset()
-        self.ds["Mesh2"] = xr.DataArray(
+        self._ds = xr.Dataset()
```

Comment:
This creates a separate  `xr.Dataset` that is stored under the `xugrid` accessor, acting like `Grid.ds` in our current implementation. It appears that currently only data variables are being store in the parent `xr.Dataset` class and our grid connectivity and other variables are stored under the `uxgrid._ds` dataset. 


Is this intentional? Do we want to separate our data variables and grid topology this way? At a high level, can we think of an `uxarray.Dataset` containing **_only_** data variables inside the inherited `xarray.Dataset` with additional functionalities and data structures  handled through accessors.

What about using the inherited `xarray.Dataset` to store **_both_** our data and grid information?

Reply:
Discussed well in #324